packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts

```diff
@@ -339,6 +346,10 @@ export class BucketDeployment extends Construct {
     // the sources actually has markers.
     const hasMarkers = sources.some(source => source.markers);
 
+    // Markers are not replaced if zip sources are not extracted, so throw an error
+    // if extraction is not wanted and sources have markers.
+    if (hasMarkers && props.extract == false) { throw new Error('Extract cannot be false if sources have markers'); }
```

Comment:
```suggestion
    if (hasMarkers && props.extract === false) { 
      throw new Error('Some sources are incompatible with extract=false; sources with deploy-time values (such as \'snsTopic.topicArn\') must be extracted.');
    }
```

My problem here is that markers is an internal property that the user is not really privy to. Getting an error that says that your source has markers is pretty cryptic. Markers are substitutes for `Ref` and `Fn::GetAtt` objects, which come from deploy-time values. So lets create an error message that reflects that.

Reply:
Thank you for the suggestion. I'll incorporate this into a better error message.